### ISSUE TITLE
bug: Screen reader does NOT announce 'required' text on mouse hover

### ISSUE THREAD

**Issue by chris-lauz** [cds-snc/gcds-components#1284]
### Prerequisites | Prérequis

- [x] I have read the [Contributing Guidelines](https://github.com/cds-snc/gcds-components/blob/main/CONTRIBUTING.md). | J’ai lu les [Lignes directrices sur les contributions](https://github.com/cds-snc/gcds-components/blob/main/CONTRIBUTING.md).
- [x] I agree to follow the [Code of Conduct](https://github.com/cds-snc/gcds-components/blob/main/CODE_OF_CONDUCT.md). | J’accepte de me conformer au [Code de conduite](https://github.com/cds-snc/gcds-components/blob/main/CODE_OF_CONDUCT.md).
- [x] I have searched for [existing issues](https://github.com/cds-snc/gcds-components/issues) that already report this problem, without success. | J’ai vérifié parmi [les tickets (issues) existants](https://github.com/cds-snc/gcds-components/issues) que ce sujet n’avait pas déjà été soumis.

### GC Design System Components Package and Version | Paquet et version des composants de Système de design GC

https://cdn.design-system.alpha.canada.ca/@cdssnc/gcds-components@0.43.1/dist/gcds/gcds.css

### Current Behavior | Comportement observé

The code snippet below shows the design-time settings for the input box as rendered in HTML (second image). It included the “required” attribute which will be a condition upon form submission.
This issue finds that when a user clicks or tabs into an input box the screen reader announces the “(required)” text along with the main label string (in this case “Latitude).
However, if a user hovers their mouse over the label, then only the “Latitude” text is announced and NOT the additional “(required)” text. They are treated as two separate strings.

<img width="975" height="213" alt="Image" src="https://github.com/user-attachments/assets/52cfcefc-2d19-4fe5-b80c-c6dad927c191" />

<img width="975" height="282" alt="Image" src="https://github.com/user-attachments/assets/dfcbac7b-f0b0-46ce-9a44-d7d00be8de52" />

### Expected Behavior | Comportement attendu

Is it reasonable to expect the "(required)" to be read aloud when the user hovers their mouse over the main/bolded portion of the label?

### System Info | Information sur le système

```shell
Microsoft Edge browser. Issue exists in alpha and release versions of GCDS.
```

### Steps to Reproduce | Étapes pour reproduire le bogue

Open any browser page containing a <gcds-input> control which has the "required" attribute set.

### Additional Information | Informations supplémentaires

Our entire application utilizing gcds controls is secured by eAccess. However, any publicly accessible application utilizing gcds controls should generate the same issue.

**Comment by ethanWallace** [comment]
Thanks for submitting this issue @chris-lauz ! It's in our backlog to take a look at this issue in the next couple of weeks.

**Comment by adriannelee** [comment]
Hi @chris-lauz, 

The way we've designed the behaviour here is similar to how other design systems do it. From @ethanWallace: 
> Currently our labels are formatted so the (required) text you see is actually hidden with aria-hidden="true" and for assistive technology the form element actually communicates that the element is required using the required attribute and not the label.

We've not come across this issue in our testing with screen reader users. However, none to my memory were using the read-aloud on hover functionality. We've booked a session with an expert screen reader user to understand what the expected behaviour is and how we might meet those expectations. Due to vacations, the session is in a couple weeks. But let us know if you have other info for us in the meantime :)

**Comment by chris-lauz** [comment]
Unclassified - Non-Classifié

Thanks for the response, Adrianne.

I will notify the team about your findings and observations so far, along with your intentions for additional tests.

Please feel free to keep PST team and with future updates.

Sincerely,

Chris Lauziere
Consultant
Fisheries and Oceans Canada | Pêches et Océans Canada

**Comment by adriannelee** [comment]
Hi @chris-lauz ! 

We had the session with the expert screenreader user in June but they unfortunately were not a user of the read-aloud on hover functionality. I reached out to the testing platform to see if they could accommodate a more specific participant criteria of a screenreader user who uses this particular functionality. They got back to me this week to let me know they won't be able to. 

So unfortunately, we don't have a good way to confirm this issue. Since this was a question rather than an 'issue', I will close the issue now but if you have new information for us, please do open it back up!
